submissions/franchukv/js-oop/index.js

```diff
@@ -0,0 +1,81 @@
+/* 
+   Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+   Code repository: https://github.com/franchukv/a-tiny-JS-world
+   Web app: https://franchukv.github.io/a-tiny-JS-world/
+*/
+
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitants {
+   constructor(species, gender, name, status, saying, friends) {
+      this.species = species;
+      this.gender = gender;
+      this.name = name;
+      this.status = status;
+      this.saying = saying;
+      this.friends = friends ? friends : `no friends`;
+      this.properties = ['species', 'gender', 'name', 'status', 'saying', 'friends',]
+   }
+
+   inhabinantPropetriesToString() {
+      return this.properties.map((property) => this[property]).join('; ');
```

Comment:
You should let child classes handle their own properties end-to-end.
Why that? Encapsulation.
Why encapsulation is important? Otherwise any change to any class properties forces us to change code in other classes. This leads to errors.